### PR TITLE
feat(pack): display promotions on offers for move or migration

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-349
| License          | BSD 3-Clause

## Description

- Display promotion on FTTH offers for migration and move pack
- Update resume table on migration and move pack: display modem rental, first year promo and rework total cost per month

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.controller.js [hunk 1]
```diff
@@ -4,11 +4,18 @@ import map from 'lodash/map';
 
 export default class TelecomPackMigrationConfirmCtrl {
   /* @ngInject */
-  constructor($q, $translate, TucPackMigrationProcess, TucToast) {
+  constructor(
+    $q,
+    $translate,
+    TucPackMigrationProcess,
+    TucToast,
+    PROMO_DISPLAY,
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.controller.js:12]: Maybe we extract this constant from angular and just import it 
```suggestion
    import {PROMO_DISPLAY} from './pack-migration-confirm.constant'
```

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 2]
```diff
@@ -67,6 +65,52 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
...
+                            <td
+                                class="text-price text-right"
+                                data-ng-bind="$ctrl.getReductionPriceFirstYear().text"
+                                data-title="{{ 'telecom_pack_migration_confirm_header_price' | translate }}"
+                            ></td>
+                            <td
+                                data-ng-bind="1"
+                                class="text-right"
+                                data-title="{{ 'telecom_pack_migration_confirm_header_quantity' | translate }}"
+                            ></td>
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:83]:
```suggestion
                            <td
                                class="text-right"
                                data-title="{{ 'telecom_pack_migration_confirm_header_quantity' | translate }}"
                            >1</td>
```
If the value is static no need for `ng-bind`

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/index.js [hunk 3]
```diff
@@ -6,10 +6,13 @@ import angularTranslate from 'angular-translate';
...
 
+import { PROMO_DISPLAY } from './pack-migration-offers.constant';
+
 const moduleName = 'ovhManagerTelecomPackMigrationOffers';
 
 angular
   .module(moduleName, [ngTranslateAsyncLoader, uiRouter, angularTranslate])
-  .component('packMigrationOffers', component);
+  .component('packMigrationOffers', component)
+  .constant('PROMO_DISPLAY', PROMO_DISPLAY);
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/index.js:16]: Not a good practice to have things named the same, especially if they contain different values. Moreover exposing an angular constant doesn't seem needed here

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html [hunk 4]
```diff
@@ -89,6 +92,20 @@
                                             data-ng-bind-html="offer.price.text | tucFormatPrice"
                                         ></small>
                                     </div>
+
+                                    <!-- Other Promotion -->
+                                    <div data-ng-if="offer.promotion && offer.promotion.id === $ctrl.PROMO_DISPLAY.PROMO_FTTH_2020" class="alert alert-info mt-2">
+                                        <div
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html:98]:
```suggestion
                                    <div data-ng-if="offer.promotion && offer.promotion.id === $ctrl.PROMO_DISPLAY.PROMO_FTTH_2020" class="oui-message oui-message_info mt-2">
                                        <div
```
?

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -136,5 +136,11 @@
...
   "telecom_pack_migration_pto_yes_not_known": "J'ai un boîtier mais je ne connais pas son numéro",
-  "telecom_pack_migration_pto_yes_not_known_description": "Avez-vous bien vérifié que vous ne disposez pas de la référence de la prise optique (PTO) ? Cela peut avoir avoir une incidence sur le délai de votre raccordement."
+  "telecom_pack_migration_pto_yes_not_known_description": "Avez-vous bien vérifié que vous ne disposez pas de la référence de la prise optique (PTO) ? Cela peut avoir avoir une incidence sur le délai de votre raccordement.",
+  "telecom_pack_migration_promotion_description": "Promotion disponible jusqu'au {{ endDate }}",
+  "telecom_pack_migration_promotion_subscription_months_offered": "{{ duration }} mois offerts",
+  "telecom_pack_migration_promotion_installation_fees_offered": "Frais de mises en service offerts",
+  "telecom_pack_migration_confirm_promotion": "Les frais d'installation et les {{ duration }} premiers mois de votre nouvel abonnement sont offerts (hors location de modem, options supplémentaires ou frais de transport).",
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json:143]: What if `duration === 1` ?

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html [hunk 6]
```diff
@@ -72,6 +74,22 @@
...
+                                            data-translate="pack_move_promotion_description"
+                                            data-translate-values="{ 'endDate' : (offer.prices.promotion.endDate | date : short) }"
+                                        >
+                                        </div>
+                                        <strong
+                                            data-ng-if="offer.prices.promotion.subscription.duration === $ctrl.PROMO_DISPLAY.P3M.key"
+                                            data-translate="pack_move_promotion_subscription_months_offered"
+                                            data-translate-values="{ 'duration' :  $ctrl.PROMO_DISPLAY.P3M.value }"
+                                        >
+                                        </strong>
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html:91]: This looks really similar to the previous one, maybe having a component could help avoiding duplication and ease maintenance

**Review by marie-j — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.constant.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const PROMO_DISPLAY = {
+  PROMO_FTTH_2020: 'FTTH_MIGRATION_AND_ADRESSMOVE_2020',
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.constant.js:2]: Shouldn't be:
```suggestion
  PROMO_FTTH_2020: 'FTTH_MIGRATION_AND_ADDRESSMOVE_2020',
```
?

Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.constant.js:2]: no it's the real name of the promotion return from API

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json (hunk 5 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/translations/Messages_fr_FR.json:143]: For the moment, there is only value for duration and is 3, I could write the message with only one value but I would like to be more generic.

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html (hunk 6 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html:91]: yes it's a good idea. I note this comment for a next rework on this part.

**Review by frenautvh — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 8]
```diff
@@ -15,9 +15,7 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
             data-translate-values="{ fromOffer : $ctrl.process.pack.offerDescription, toOffer : $ctrl.process.selectedOffer.description }"
         ></p>
 
-        <div
-            data-ng-if="$ctrl.choosedAdditionalOptions.length > 0 || ($ctrl.process.selectedOffer.options.gtr_ovh && $ctrl.process.selectedOffer.options.gtr_ovh.selected)"
-        >
+        <div>
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:18]: **thought:** The following `div` looks like useless. Could be removed on separate Pull Request.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html [hunk 9]
```diff
@@ -89,6 +92,21 @@
                                             data-ng-bind-html="offer.price.text | tucFormatPrice"
                                         ></small>
                                     </div>
+
+                                    <!-- Other Promotion -->
+                                    <div data-ng-if="offer.promotion && offer.promotion.id === $ctrl.PROMO_DISPLAY.PROMO_FTTH_2020" class="oui-message oui-message_info mt-2">
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html:97]: **thought:** Looks like a [`<oui-message type="info"`](https://ovh.github.io/ovh-ui-kit/?path=/story/design-system-components-message-webcomponent--default)

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html [hunk 10]
```diff
@@ -141,6 +159,24 @@
                                         ng-if="offer.installFees"
                                         data-ng-bind-html="offer.installFees.text | tucFormatPrice"
                                     ></span>
+
+                                    <!-- Other promotion -->
+                                    <div data-ng-if="offer.promotion.id=='FTTH_MIGRATION_AND_ADRESSMOVE_2020'" class="oui-message oui-message_info mt-2">
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.html:164]: **thought:** Looks like a [`<oui-message type="info"`](https://ovh.github.io/ovh-ui-kit/?path=/story/design-system-components-message-webcomponent--default)

File: packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html [hunk 11]
```diff
@@ -72,6 +74,23 @@
                                         data-ng-bind-html="offer.prices.price.price.text | tucFormatPrice"
                                     ></span>
                                 </div>
+
+                                <!-- Other promotion -->
+                                <div
+                                    data-ng-if="offer.prices.promotion && offer.prices.promotion.id === $ctrl.PROMO_DISPLAY.PROMO_FTTH_2020" class="oui-message oui-message_info mt-2">
```
Comment (antleblanc) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.html:80]: **thought:** Looks like a [`<oui-message type="info"`](https://ovh.github.io/ovh-ui-kit/?path=/story/design-system-components-message-webcomponent--default)

**Review by antleblanc — APPROVED** [review]
Changes looks good to me.

Added some comments that can be handled as a separate Pull Request (in addition to the previous Marie's comments).

Thanks!

**Comment by Steffy29** [comment]
@antleblanc OK, I add your comments to the created task for component that list offers available for migration and move pacK. Thanks

**Comment by antleblanc** [comment]
@Steffy29 In order to be able to merge this Pull Request, could you please resolve existing conflict?

Thank you!

**Comment by Steffy29** [comment]
@antleblanc Conflicts are resolved.

**Review by antleblanc — APPROVED** [review]
